packages/core/src/repositories/connectedEvaluationsRepository.ts

```diff
@@ -0,0 +1,147 @@
+
+    return Result.ok(result[0]!)
+  }
+
+  async getConnectedDocumentsWithMetadata(
+    evaluationId: number,
+  ): Promise<TypedResult<ConnectedDocumentWithMetadata[], LatitudeError>> {
+    const documentVersionsScope = new DocumentVersionsRepository(
+      this.workspaceId,
+      this.db,
+    )
+
+    const aggregatedMetadata = this.db
+      .select({
+        evaluationId: evaluationResults.evaluationId,
+        documentUuid: documentLogs.documentUuid,
+        evaluationLogs: count(evaluationResults.id).as('evaluation_logs'),
+        totalTokens: sum(providerLogs.tokens).as('total_tokens'),
+        costInMillicents: sum(providerLogs.cost_in_millicents).as(
+          'cost_in_millicents',
+        ),
+      })
+      .from(evaluationResults)
+      .leftJoin(
+        documentLogs,
+        eq(documentLogs.id, evaluationResults.documentLogId),
+      )
+      .leftJoin(
+        providerLogs,
+        eq(providerLogs.id, evaluationResults.providerLogId),
+      )
+      .groupBy(documentLogs.documentUuid, evaluationResults.evaluationId)
+      .as('aggregated_metadata')
+
+    const latestDocumentVersions = this.db
+      .select({
+        documentUuid: documentVersionsScope.scope.documentUuid,
+        latestMergedAt: max(documentVersionsScope.scope.mergedAt).as(
+          'latest_merged_at',
+        ),
+      })
+      .from(documentVersionsScope.scope)
+      .where(isNotNull(documentVersionsScope.scope.mergedAt)) // Only account for merged documents
+      .groupBy(documentVersionsScope.scope.documentUuid)
+      .as('latest_document_versions')
+
+    const result = await this.db
+      .select({
+        ...documentVersionsScope.scope._.selectedFields,
+        evaluationLogs:
+          sql<number>`COALESCE(${aggregatedMetadata.evaluationLogs}, 0)`
+            .mapWith(Number)
+            .as('evaluation_logs'),
+        totalTokens: sql<number>`COALESCE(${aggregatedMetadata.totalTokens}, 0)`
+          .mapWith(Number)
+          .as('total_tokens'),
+        costInMillicents:
+          sql<number>`COALESCE(${aggregatedMetadata.costInMillicents}, 0)`
+            .mapWith(Number)
+            .as('cost_in_millicents'),
+      })
+      .from(this.scope)
+      .innerJoin(
+        latestDocumentVersions,
+        eq(this.scope.documentUuid, latestDocumentVersions.documentUuid),
+      )
+      .innerJoin(
+        documentVersionsScope.scope,
+        and(
+          eq(this.scope.documentUuid, documentVersionsScope.scope.documentUuid),
+          eq(
+            documentVersionsScope.scope.mergedAt,
+            latestDocumentVersions.latestMergedAt,
+          ),
+        ),
+      )
+      .leftJoin(
+        aggregatedMetadata,
+        eq(this.scope.evaluationId, aggregatedMetadata.evaluationId),
+      )
+      .where(eq(this.scope.evaluationId, evaluationId))
+
+    return Result.ok(result.filter((r) => r.deletedAt == null)) // Only show non-removed documents
+  }
```

Comment:
this merits a test case tbh

Reply:
😭

Reply:
Venga que cursor te los escribe solo casi

Reply:
☝️ Rewrote the queries to add a modal value, and simplify the overall query complexity. Working on tests now!

Reply:
Done! Had to modify some factories and a few other functions, but the test env is a bit cleaner now!